src/test/java/it/gov/pagopa/debtposition/scheduler/ExpiredPositionsSchedulerTest.java

```diff
@@ -201,5 +204,86 @@ void manualChangeDebtPositionStatusToExpiredAfterDueDate()
+		mvc.perform(post("/organizations/SCHEDULEEXPANDUPD_12345678901/debtpositions/12345678901IUPDMOCK3/publish")
+				.contentType(MediaType.APPLICATION_JSON)).andExpect(status().isOk());
+
+		// verifico che lo stato sia stato effettivamente aggiornato a PUBLISHED 
+		mvc.perform(get("/organizations/SCHEDULEEXPANDUPD_12345678901/debtpositions/12345678901IUPDMOCK3")
+				.contentType(MediaType.APPLICATION_JSON)).andExpect(status().isOk())
+		.andExpect(content().contentType(MediaType.APPLICATION_JSON))
+		.andExpect(MockMvcResultMatchers.jsonPath("$.status")
+				.value(DebtPositionStatus.PUBLISHED.toString()))
+		.andExpect(MockMvcResultMatchers.jsonPath("$.publishDate").isNotEmpty());
+
+		// introduco un ritardo in modo da far scadere la validity date
+		LocalDateTime currentDatePlusSeconds = LocalDateTime.now(ZoneOffset.UTC).plus(5, ChronoUnit.SECONDS);
+		Awaitility.await().until(() -> LocalDateTime.now(ZoneOffset.UTC).isAfter(currentDatePlusSeconds));
+
+		// lancio il batch per consentire il passaggio di stato
+		expiredPositionsScheduler.changeDebtPositionStatusToValid();
+
+		Awaitility.await()
+		.atMost(3, TimeUnit.SECONDS)
+		.pollInterval(15, TimeUnit.MILLISECONDS)
+		.until(() -> expiredPositionsScheduler.getThreadOfExecution() != null);
+
+		// dopo che il batch ha terminato verifico che lo stato sia effettivamente passato a VALID
+		mvc.perform(get("/organizations/SCHEDULEEXPANDUPD_12345678901/debtpositions/12345678901IUPDMOCK3")
+				.contentType(MediaType.APPLICATION_JSON)).andExpect(status().isOk())
+		.andExpect(content().contentType(MediaType.APPLICATION_JSON))
+		.andExpect(MockMvcResultMatchers.jsonPath("$.status")
+				.value(DebtPositionStatus.VALID.toString()));
+
+		// introduco un nuovo ritardo in modo da far scadere la due date
+		LocalDateTime newCurrentDatePlusSeconds = LocalDateTime.now(ZoneOffset.UTC).plus(5, ChronoUnit.SECONDS);
+		Awaitility.await().until(() -> LocalDateTime.now(ZoneOffset.UTC).isAfter(newCurrentDatePlusSeconds));
+
+		// lancio il batch per consentire il passaggio di stato
+		expiredPositionsScheduler.changeDebtPositionStatusToExpired();
+
+		// attendo che il thread asincrono sia attivo
+		Awaitility.await()
+		.atMost(3, TimeUnit.SECONDS)
+		.pollInterval(15, TimeUnit.MILLISECONDS)
+		.until(() -> expiredPositionsScheduler.getThreadOfExecution() != null);
+
+		// verifico che lo stato sia passato ad EXPIRED
+		mvc.perform(get("/organizations/SCHEDULEEXPANDUPD_12345678901/debtpositions/12345678901IUPDMOCK3")
+				.contentType(MediaType.APPLICATION_JSON)).andExpect(status().isOk())
+		.andExpect(content().contentType(MediaType.APPLICATION_JSON))
+		.andExpect(MockMvcResultMatchers.jsonPath("$.status")
+				.value(DebtPositionStatus.EXPIRED.toString()));
+		
+		// aggiorno la posizione debitoria con un body che non contiene la 'validity date' e ne richiedo la pubblicazione
+		PaymentPositionDTO pp7 = DebtPositionMock.getMock7();
+		pp7.setValidityDate(null);
```

Comment:
Is another test with validityDate prior to now useful? Could be an input of this method with 2 values `(null, now-10m)`